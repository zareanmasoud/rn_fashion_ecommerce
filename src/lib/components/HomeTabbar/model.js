// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
