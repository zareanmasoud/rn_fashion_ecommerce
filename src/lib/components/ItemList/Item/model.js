// @flow

export type ItemModel = {
  id: string,
  photo: string,
};
