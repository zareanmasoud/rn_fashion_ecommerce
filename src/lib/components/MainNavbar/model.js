// @flow

export type Button = {
  action: Function,
  src: string,
};
