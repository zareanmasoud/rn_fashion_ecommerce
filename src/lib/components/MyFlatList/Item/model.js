// @flow

export type Data = {
  id: string,
  photo: string,
};
