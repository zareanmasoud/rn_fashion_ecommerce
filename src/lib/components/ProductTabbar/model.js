// @flow

export type Specs = {
  action: Function,
  src: string,
  text: string,
};
