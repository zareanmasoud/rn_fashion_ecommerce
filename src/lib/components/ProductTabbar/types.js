// @flow

export type Button = {
  handlePress: Function,
  src: string,
  text: string,
};
