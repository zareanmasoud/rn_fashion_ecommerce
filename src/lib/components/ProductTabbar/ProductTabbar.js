import React from 'react';
import {View} from 'react-native';
import TabbarButton from 'lib/components/TabbarButton';
import AddToCartButton from 'lib/components/AddToCartButton';
import styles from './styles';
import type {Button} from './types';

type Props = {
  buttons: Button[],
  addToCart: Button,
};

const ProductTabbar = ({buttons, addToCart}: Props) => {
  const TabbarButtons = buttons.map(button => (
    <TabbarButton {...button} key={button.id} mode="product" />
  ));

  return (
    <View style={styles.container}>
      {TabbarButtons}
      <AddToCartButton {...addToCart} />
    </View>
  );
};

export default ProductTabbar;
