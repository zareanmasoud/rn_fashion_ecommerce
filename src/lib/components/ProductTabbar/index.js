import ProductTabbar from './ProductTabbar';

export {ProductTabbar as default};
