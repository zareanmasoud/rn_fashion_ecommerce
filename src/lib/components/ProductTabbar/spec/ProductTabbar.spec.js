import React from 'react';
import {shallow} from 'enzyme';
import ProductTabbar from '../ProductTabbar';
import model from './model';

describe('ProductTabbar Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(
        <ProductTabbar buttons={model.buttons} addToCart={model.addToCart} />,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
