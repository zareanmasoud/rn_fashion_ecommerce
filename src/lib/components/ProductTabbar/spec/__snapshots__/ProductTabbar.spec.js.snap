// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTabbar Component Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 1,
      "flexDirection": "row",
      "height": 107.14285714285714,
      "justifyContent": "space-between",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 0,
    }
  }
>
  <TabbarButton
    handlePress={[Function]}
    id="0"
    key="0"
    mode="product"
    src={
      Object {
        "process": [Function],
      }
    }
    text=""
  />
  <TabbarButton
    handlePress={[Function]}
    id="1"
    key="1"
    mode="product"
    src={
      Object {
        "process": [Function],
      }
    }
    text=""
  />
  <AddToCartButton
    handlePress={[Function]}
    id="2"
    src={
      Object {
        "process": [Function],
      }
    }
    text=""
  />
</View>
`;
